src/wallet/wallet.cpp

```diff
@@ -2991,6 +2991,165 @@ std::string CWallet::GetWalletHelpString(bool showDebug)
     return strUsage;
 }
 
+CWallet* CWallet::InitLoadWallet(bool fDisableWallet, const std::string& strWalletFile, std::string& warningString, std::string& errorString)
+{
+    // needed to restore wallet transaction meta data after -zapwallettxes
+    std::vector<CWalletTx> vWtx;
+
+    if (GetBoolArg("-zapwallettxes", false)) {
+        uiInterface.InitMessage(_("Zapping all transactions from wallet..."));
+
+        CWallet *tempWallet = new CWallet(strWalletFile);
+        DBErrors nZapWalletRet = tempWallet->ZapWalletTx(vWtx);
+        if (nZapWalletRet != DB_LOAD_OK) {
+            errorString = _("Error loading wallet.dat: Wallet corrupted");
+            uiInterface.InitMessage(_("Error loading wallet.dat: Wallet corrupted"));
+            return NULL;
+        }
+
+        delete tempWallet;
+        tempWallet = NULL;
+    }
+
+    uiInterface.InitMessage(_("Loading wallet..."));
+
+    int64_t nStart = GetTimeMillis();
+    bool fFirstRun = true;
+    CWallet *walletInstance = new CWallet(strWalletFile);
+    DBErrors nLoadWalletRet = walletInstance->LoadWallet(fFirstRun);
+    if (nLoadWalletRet != DB_LOAD_OK)
+    {
+        if (nLoadWalletRet == DB_CORRUPT)
+            errorString += _("Error loading wallet.dat: Wallet corrupted") + "\n";
```

Comment:
It is ugly but was already the case in the previous version. This literally moves the code. Let's leave improvements here to later pulls.